packages/xo-server/src/xo-mixins/backups-ng/index.js

```diff
@@ -150,6 +150,38 @@ export default class BackupNg {
         // different than the VMs in the job itself.
         let vmIds = data?.vms ?? extractIdsFromSimplePattern(vmsPattern)
         if (vmIds === undefined) {
+          let poolIds = extractIdsFromSimplePattern({ id: vmsPattern.$pool })
+          if (poolIds === undefined && Array.isArray(vmsPattern.$pool?.__and)) {
+            poolIds = []
+            vmsPattern.$pool.__and.forEach(pattern => {
+              const ids = extractIdsFromSimplePattern({ id: pattern })
+              if (ids !== undefined) {
+                poolIds = [...poolIds, ids]
+              }
+            })
+          }
```

Comment:
This code is not easy to understand, it could be commented.

Also, I believe there is an error.